### PR TITLE
Chore/nx migration

### PR DESCRIPTION
**Pull request by mariojsnunes**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

_What this PR does_

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: apps/community-platform/index.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      rel="preconnect"
+      href="https://firebasestorage.googleapis.com"
+      crossorigin
+    />
+    <noscript><link rel="stylesheet" href="path/to/stylesheet.css" /></noscript>
+    <meta name="theme-color" content="#000000" />
+    <meta property="og:title" content="Community Platform" />
+    <meta
+      property="og:description"
+      content="A series of tools for the Precious Plastic community to collaborate around the world. Connect, share and meet each other to tackle plastic waste."
```
Comment (benfurber) [inline apps/community-platform/index.html:20]: Should be generic

File: apps/community-platform/index.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    />
+    <meta property="og:image" content="./social-image.jpg" />
+    <meta property="og:url" content="https://community.preciousplastic.com" />
```
Comment (benfurber) [inline apps/community-platform/index.html:23]: Should be generic

File: apps/community-platform/index.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    <meta name="twitter:title" content="Community Platform" />
+    <meta
+      name="twitter:description"
+      content="A series of tools for the Precious Plastic community to collaborate around the world. Connect, share and meet each other to tackle plastic waste."
```
Comment (benfurber) [inline apps/community-platform/index.html:27]: Should be generic

File: apps/community-platform/src/pages/Howto/Content/Howto/Howto.tsx [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
-import { seoTagsUpdate } from 'src/utils/seo'
+} from '@onearmy.apps/components'
+import { IModerationStatus } from '@onearmy.apps/shared'
+import { observer } from 'mobx-react'
 import { Box } from 'theme-ui'
 
+import { trackEvent } from '../../../../common/Analytics'
+import { useCommonStores } from '../../../../common/hooks/useCommonStores'
+import { Breadcrumbs } from '../../../../pages/common/Breadcrumbs/Breadcrumbs'
+import { seoTagsUpdate } from '../../../../utils/seo'
```
Comment (benfurber) [inline apps/community-platform/src/pages/Howto/Content/Howto/Howto.tsx:17]: General point, just picking this as an example. The depth of these relative paths is annoying. No other option?

File: .eslintrc.json [hunk 5]
```diff
@@ -14,8 +14,6 @@
     "eslint:recommended",
     "plugin:react/recommended", // Uses the recommended rules from @eslint-plugin-react
     "plugin:@typescript-eslint/recommended", // Uses the recommended rules from the @typescript-eslint/eslint-plugin
-    "plugin:import/recommended",
-    "plugin:import/typescript",
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply .eslintrc.json:18]: the import plugin was preventing nx from running. (maybe due to commonjs?)

File: .eslintrc.json [hunk 6]
```diff
@@ -40,15 +38,6 @@
         "accessorPairPositioning": "getThenSet"
       }
     ],
-    "import/no-duplicates": "error",
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply .eslintrc.json:43]: the import plugin was preventing nx from running. (maybe due to commonjs?)

File: apps/community-platform/index.html [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    <link id="favicon" rel="shortcut icon" href="./favicon.ico" />
+    <!--
+      Notice the use of ./public in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+      Unlike "/favicon.ico" or "favicon.ico", "./favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>Precious Plastic Community</title>
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply apps/community-platform/index.html:45]: I think we have a script that replaces this before serving, but it's something to confirm.

File: apps/community-platform/src/pages/Howto/Content/Howto/Howto.tsx (hunk 4 above)
Comment (mariojsnunes) [reply apps/community-platform/src/pages/Howto/Content/Howto/Howto.tsx:17]: Should be fixed by splitting up the code in libraries. ex:
`import { Breadcrumbs } from '@onearmy.apps/shared'`

**Comment by mariojsnunes** [comment]
thanks @benfurber will consider all your comments on the next try.

Important info from discord:

> The NX migration requires more structural changes than anticipated. Mostly due to the e2e tests. Ultimately, this means a better project structure.
> Before detailing the new structure, there are two concepts to understand: apps and libs.
> Libs can be standalone like our UI components. Or can depend on other libs.
> Apps can reference libs. But nothing should reference apps.
> These concepts ensure we keep a good project structure and avoid complex or hard to maintain dependencies. It is enforced by a 'module boundaries' eslint rule.
>
> So why does it require more changes than I anticipated? 
> Our cypress tests will exist in an app (community-platform-e2e). Which means we cannot reference the main community-platform app. And currently we reference it to access constant variables.
> To solve this the community-platform app must be splitted in multiple libs (a good NX practice).
> 
> What does it look like?
> apps:
> community-platform
> community-platform-e2e
> 
> libs:
> components
> shared
> howtos
> research
> questions
> profile
> maps
> ...
> 

Abandoning this PR as I found a few components that are coupling things up too much:

- Breadcrumbs
- useCommonStores
- deprecated CategoriesSelect used by Howtos
- likely more
- 

fixing them first before proceeding with the migration (as fixing them means changing code, not just moving things around)

**Pull request closed without merging**